docs/transpiler.md

```diff
@@ -111,12 +111,13 @@ then `load()` from your own macro rather than from `@aspect_rules_ts`.
 
 ## Macro expansion
 
-When `transpiler` is used, then the `ts_project` macro expands to these targets:
+When `transpiler` or `declaration_transpiler` is used, then the `ts_project` macro expands to these targets:
```

Comment:
@gregmagolan I updated this doc since the last review

Reply:
Reading the code I think there are cases when the macro does not expand to all of the targets tho the exact matrix is hard to follow. When no_emit is True is one of those cases I see.

```
if no_emit or not declaration_transpiler:
   ...
elif not no_emit and declaration:
   ...
```

is particularly confusing

Reply:
Speaking of no_emit, that flag is particularly confusing if there is a custom `transpiler` or `declaration_transpiler` set since the docstring for that parameter is no longer accurate as Bazel is expecting outputs, just not from the `{name}_tsc` target.

Reply:
It looks like `transpiler` and `declaration_transpiler` are ignored if `no_emit` is True so should probably help the user out and `fail` on that condition.